src/redux/actions/authActions.js

```diff
@@ -0,0 +1,143 @@
+import fetch from 'cross-fetch';
+
+export const SIGNUP = 'SIGN_UP';
+export const SIGNUP_REQUEST = 'REQUEST_SIGNUP';
+export const SIGNUP_FAILURE = 'SIGNUP_FAILURE';
+export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
+export const LOGOUT = 'LOGOUT';
+export const LOGIN = 'LOGIN';
+export const LOGIN_REQUEST = 'REQUEST_LOGIN';
+export const LOGIN_FAILURE = 'LOGIN_FAILURE';
+export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
+export const LOGOUT_REQUEST = 'LOGOUT_REQUEST';
+export const LOGOUT_FAILURE = 'LOGOUT_FAILURE';
+export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
+
+function requestSignup(user) {
+  return {
+    type: SIGNUP_REQUEST,
+    user: user,
+    receivedAt: Date.now()
+  };
+}
+
+function signupSuccess(user, json) {
+  return {
+    type: SIGNUP_SUCCESS,
+    user: json,
+    receivedAt: Date.now()
+  };
+}
+
+function signupFailure() {
+  return {
+    type: SIGNUP_FAILURE,
+    user: null,
+    receivedAt: Date.now()
+  };
+}
+
+function requestLogin(user) {
+  return {
+    type: LOGIN_REQUEST,
+    user: user,
+    receivedAt: Date.now()
+  };
+}
+
+function loginSuccess(user, json) {
+  return {
+    type: LOGIN_SUCCESS,
+    user: json,
```

Comment:
There are two parameters here—why is user mapped to json, and the parameter user is thrown away? This is true for signupSuccess as well. I might just not be following what's going on.

Reply:
The only param should be user. Not sure what I was thinking here... Will update. Thanks for catching this.